docker/Dockerfile

```diff
@@ -1,44 +1,39 @@
-FROM python:3.7.2-alpine3.8
-
-ENV PYTHONUNBUFFERED=1
-ENV PYTHONDONTWRITEBYTECODE=1
+FROM python:3.7.3-slim as base
 ENV PATH="/root/.poetry/bin:$PATH"
 ENV FLASK_APP=search
-
 WORKDIR /srv/app
-
-RUN rm -rf /var/cache/apk/* && \
-    rm -rf /tmp/* && \
-    apk update && \
-    apk add --no-cache \
-    bash \
-    grep \
-    libxslt-dev
-
-RUN apk add --no-cache --virtual .build-dependencies \
-    curl \
-    && \
-    curl -sSL https://raw.githubusercontent.com/sdispater/poetry/master/get-poetry.py | python && \
-    poetry config settings.virtualenvs.create false && \
-    apk del .build-dependencies
-
 COPY ./pyproject.toml ./poetry.lock ./
-
-ARG env
-ENV FLASK_ENV=${env}
-
-ARG dev
-ARG simulate_removal
-RUN apk add --no-cache --virtual .build-dependencies \
-    gcc \
-    git \
-    linux-headers \
-    musl-dev \
-    && \
-    poetry install ${dev} && \
-    apk del ${simulate_removal} .build-dependencies
-
 COPY search/ search/
-
+RUN set -ex \
+    && apt-get update -yq \
+    && apt-get install -yq --no-install-recommends curl \
+    && curl -sSL https://raw.githubusercontent.com/sdispater/poetry/master/get-poetry.py | python \
+    && poetry config settings.virtualenvs.create false \
+    && apt-get remove --purge --autoremove -yq curl
+RUN set -ex \
```

Comment:
why not a single `RUN` command with the one before?
If keeping them separate, the idiom is to insert a new `apt-get update`, otherwise the first `RUN` being cached may provide outdated index files (referencing package versions not available anymore) to the `apt-get install` in this one.

Reply:
Simmetrically, `rm -rf /var/lib/apt/lists/*` should be included in the previous one so that index files are not left in any Docker image layer.

Reply:
My original thinking was, by running these separately, poetry would not be reinstalled when dependencies change. Now looking at this with fresh eyes, I think it makes sense to put these two operations into a single RUN command.